test/unit/behaviors.spec.js

```diff
@@ -410,509 +508,600 @@ describe('Behaviors', function() {
+          onDestroy: onDestroyStub
         })
       };
-      Marionette.Behaviors.behaviorsLookup = this.behaviors;
 
-      this.View = Marionette.View.extend({
+      this.sinon.stub(Marionette.Behaviors, 'behaviorsLookup', function() {
+        return behaviors;
+      });
+
+      let FooView = View.extend({
         template: _.template('foo'),
         behaviors: {foo: {}}
       });
 
       this.setFixtures('<div id="region"></div>');
-      this.region = new Backbone.Marionette.Region({el: '#region'});
-      this.view = new this.View();
+      let region = new Region({el: '#region'});
+      let fooView = new FooView();
 
-      this.region.show(this.view);
-      this.region.empty();
+      region.show(fooView);
+      region.empty();
     });
 
     it('behavior onAttach is called once', function() {
-      expect(this.onAttachStub).to.have.been.calledOnce;
+      expect(onAttachStub).to.have.been.calledOnce;
     });
 
     it('behavior onClose is called once', function() {
-      expect(this.onDestroyStub).to.have.been.calledOnce;
+      expect(onDestroyStub).to.have.been.calledOnce;
+    });
+
+    afterEach(function() {
+      Marionette.Behaviors.behaviorsLookup.restore();
     });
   });
 
   describe('behavior instance events', function() {
+    let listenToChangeStub;
+    let onFooStub;
+
     beforeEach(function() {
-      var suite = this;
-      this.listenToChangeStub = this.sinon.stub();
-      this.onFooStub = this.sinon.stub();
-      this.model = new Backbone.Model();
+      let model = new Backbone.Model();
 
-      this.FooBehavior = Marionette.Behavior.extend({
+      listenToChangeStub = this.sinon.stub();
+      onFooStub = this.sinon.stub();
+
+      let FooBehavior = Behavior.extend({
         initialize: function() {
-          this.listenTo(suite.model, 'change', suite.listenToChangeStub);
-          this.on('foo', suite.onFooStub);
+          this.listenTo(model, 'change', listenToChangeStub);
+          this.on('foo', onFooStub);
         }
       });
 
-      this.View = Marionette.View.extend({
-        behaviors: {foo: {behaviorClass: this.FooBehavior}}
+      let FooView = View.extend({
+        behaviors: {foo: {behaviorClass: FooBehavior}}
       });
 
-      this.view = new this.View();
-      this.view.destroy();
-      this.model.set('bar', 'baz');
-      this.view.triggerMethod('foo');
+      let fooView = new FooView();
+      fooView.destroy();
+      model.set('bar', 'baz');
+      fooView.triggerMethod('foo');
     });
 
     it('should unbind listenTo on destroy', function() {
-      expect(this.listenToChangeStub).not.to.have.been.calledOnce;
+      expect(listenToChangeStub).not.to.have.been.calledOnce;
     });
 
     it('should still be bound to "on" on destroy', function() {
-      expect(this.onFooStub).to.have.been.calledOnce;
+      expect(onFooStub).to.have.been.calledOnce;
     });
   });
 
   describe('behavior model events', function() {
+    let handleModelChangeStub;
```

Comment:
This test is too big, I'll try to split it up. One more think, I'm still doubt about
```
let foo;
let bar;
let baz;
```
and
```
let foo, bar, baz;
```